### PR TITLE
NOISSUE: app compartment (rev3)

### PR DESCRIPTION
**Pull request by cyraxred**
* added app compartment and test server for LMN
* extended conveyor's caching to include OnlinePopulation to pulseSlot
* isolation of conveyor's slotmachine from pulse slots
* redesign of cryptography scheme 
* improved / simplified app compartment config and impose
* additional injection containers

### CODE REVIEW

File: ledger-core/ledger/server/lmnapp/lmn_app_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+package lmnapp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/insolar/assured-ledger/ledger-core/instrumentation/insconveyor"
+)
+
+func TestStartCtm(t *testing.T) {
```
Comment (ivanshibitov) [inline ledger-core/ledger/server/lmnapp/lmn_app_test.go:15]: What does this test check?

File: ledger-core/crypto/legacyadapter/digest_256.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+func (pd *Sha3Digester256) DigestData(reader io.Reader) cryptkit.Digest {
+	hasher := pd.scheme.IntegrityHasher()
+
+	_, err := io.Copy(hasher, reader)
+	if err != nil {
+		panic(err)
+	}
+
+	bytes := hasher.Sum(nil)
+	bits := longbits.NewBits256FromBytes(bytes[:32])
```
Comment (ivanshibitov) [inline ledger-core/crypto/legacyadapter/digest_256.go:37]: I would move '32' to constant and refuse it here and below

File: ledger-core/ledger/server/buildersvc/service.go [hunk 3]
```diff
@@ -34,18 +34,22 @@ type AppendFuture interface {
 }
 
 type Service interface {
-	CreatePlash(node.ShortNodeID, pulse.Range, jet.Tree, census.OnlinePopulation) (PlashAssistant, []jet.ExactID)
+	CreatePlash(pulse.Range, jet.Tree, census.OnlinePopulation) (PlashAssistant, []jet.ExactID)
```
Comment (ivanshibitov) [inline ledger-core/ledger/server/buildersvc/service.go:37]: There is no word 'Plash' in Cambridge dictionary =(

Comment (cyraxred) [reply ledger-core/ledger/server/buildersvc/service.go:37]: https://www.merriam-webster.com/dictionary/plash

File: ledger-core/crypto/legacyadapter/digest_256.go (hunk 2 above)
Comment (cyraxred) [reply ledger-core/crypto/legacyadapter/digest_256.go:37]: 1. this is legacy adapter to be removed
2. a string constant for 256 bits = 32 bytes ... does it make any sense?

File: ledger-core/ledger/server/lmnapp/lmn_app_test.go (hunk 1 above)
Comment (cyraxred) [reply ledger-core/ledger/server/lmnapp/lmn_app_test.go:15]: "start compartment"
it runs initial part of LNM compartment. It is skipped, as LMN init logic of SMs isn't ready.

File: ledger-core/ledger/server/buildersvc/service.go (hunk 3 above)
Comment (cyraxred) [reply ledger-core/ledger/server/buildersvc/service.go:37]: https://dictionary.cambridge.org/dictionary/english/plush

you have to look for full english dictionary, not for essential one

Comment (cyraxred) [reply ledger-core/ledger/server/buildersvc/service.go:37]: free version of Cambridge dictionary seems to be limited ...

**Pull request merged**